src/common/modules/AutocorrectHandler.js

```diff
@@ -0,0 +1,206 @@
+        let length = 0;
+        let index = x.length;
+
+        for (const y in autocorrections) {
+            if (x === y) {
+                continue;
+            }
+            const aindex = x.indexOf(y);
+            if (aindex >= 0) {
+                if (aindex < index) {
+                    index = aindex;
+                    length = y.length;
+                } else if (aindex === index && y.length > length) {
+                    length = y.length;
+                }
+            }
+        }
+
+        if (length > 0) {
+            length = x.length - (index + length);
+            if (length > 1) {
+                antipatterns.push(x.slice(0, -(length - 1)));
+            }
+        }
+    }
+    antipatterns = antipatterns.filter((item, pos) => antipatterns.indexOf(item) === pos);
+    console.log("Do not autocorrect for these patterns", antipatterns);
+
+    for (const [index, symbol] of antipatterns.entries()) {
+        antipatterns[index] = symbol.replace(regExSpecialChars, "\\$&");
+    }
+
+    symbolpatterns = new RegExp(`(${symbolpatterns.join("|")})$`);
+    antipatterns = new RegExp(`(${antipatterns.join("|")})$`);
+}
+
+/**
+ * On error.
+ *
+ * @param {string} error
+ * @returns {void}
+ */
+function onError(error) {
+    console.error(`Error: ${error}`);
+}
+
+/**
+ * Set autocorrect settings.
+ *
+ * @param {Object} autocorrect
+ * @returns {void}
+ */
+function setSettings(autocorrect) {
+    settings.autocorrectEmojis = autocorrect.autocorrectEmojis;
+    settings.autocorrectEmojiShortcodes = autocorrect.autocorrectEmojiShortcodes;
+    settings.autocomplete = autocorrect.autocompleteEmojiShortcodes;
+
+    applySettings();
+}
+
+/**
+ * Send autocorrect settings to content scripts.
+ *
+ * @param {Object} autocorrect
+ * @returns {void}
+ */
+function sendSettings(autocorrect) {
+    setSettings(autocorrect);
+
+    browser.tabs.query({}).then((tabs) => {
+        for (const tab of tabs) {
+            browser.tabs.sendMessage(
+                tab.id,
+                {
+                    "type": COMMUNICATION_MESSAGE_TYPE.AUTOCORRECT_CONTENT,
+                    "autocomplete": settings.autocomplete,
+                    "autocorrections": autocorrections,
+                    "longest": longest,
+                    "symbolpatterns": symbolpatterns,
+                    "antipatterns": antipatterns,
+                    "emojiShortcodes": emojiShortcodes
+                }
+            ).catch(onError);
+        }
+    }).catch(onError);
+}
+
+/**
+ * Init autocorrect module.
+ *
+ * @public
+ * @returns {void}
+ */
+export async function init() {
+    const autocorrect = await AddonSettings.get("autocorrect");
+
+    for (const key in emojiMart.emojiIndex.emojis) {
+        const emoji = emojiMart.emojiIndex.emojis[key];
+        if (!emoji.native) {
+            /* for (const key in emoji) {
+				const aemoji = emoji[key];
+				emojiShortcodes[aemoji.colons] = aemoji.native;
+			} */
```

Comment:
dead code?

Reply:
No, but it is incorrectly indented. This block is to support autocorrecting `:colon:` shortcodes with the skin tone modifiers. The feature would need some additional work, which I think is out of scope for this PR and why I commented it out. I can remove this block if you want...

Reply:
Well, it's okay, just add a `//TODO` then and link to a new issue or so, where you'd track progress on the new feature.

Reply:
OK, I will just remove it for now... I of course cannot create any new issues or PRs related to the autocorrect feature until this PR is merged and as you may recall, I already have [four more PRs](https://github.com/rugk/awesome-emoji-picker/issues/106#issuecomment-712144781) lined up waiting on this.